### PR TITLE
Automate releases

### PR DESCRIPTION
**Pull request by karls**
This PR builds on #21 and introduces `.github/workflows/release.yml` which automates building the library, creating a release and publishing a new version on PyPI. This happens _only_ on `main` and _only_ when all tests have passed.

Note: this PR's branch was created from the branch for #21, which is why it includes commits from there. This should be ready to go once #21 is reviewed and merged.

### CODE REVIEW

**Comment by adamlogic** [comment]
@karls FYI I changed the base branch of this PR to `migrate-to-poetry` so this PR only includes the commits exclusive to this branch. GitHub will automatically change the base branch to `main` when `migrate-to-poetry` is merged.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-root
+        if: ${{ steps.cache-deps.outputs.cache-hit != 'true' }}
+
+      - name: Install library
+        run: poetry install --no-interaction
+
+      - name: Publish to PyPI
+        run: poetry publish --build --dry-run
```
Comment (adamlogic) [inline .github/workflows/release.yml:69]: Were you thinking we'd remove `--dry-run` before merging, after I have the secret in place?

Comment (karls) [reply .github/workflows/release.yml:69]: I think I'd like to see the workflow run successfully with `--dry-run` and with the secret in place before removing it. If you're OK with that, we could merge _with_ `--dry-run` and verify that everything works, then remove the flag when we're ready for a new release.

**Comment by karls** [comment]
> GitHub will automatically change the base branch to `main` when `migrate-to-poetry` is merged.

I did not know that! Good stuff!

File: .github/workflows/release.yml (hunk 1 above)
Comment (adamlogic) [reply .github/workflows/release.yml:69]: Sounds good! 👍

**Review by adamlogic — APPROVED** [review]

**Pull request merged**